src/shapesdemo/ShapePublisher.cpp

```diff
@@ -83,3 +92,10 @@ void ShapePublisher::on_offered_deadline_missed(
 {
     m_mainWindow->addMessageToOutput(QString("Offered deadline missed"));
 }
+
+void ShapePublisher::on_liveliness_lost(
+        Publisher*,
+        const LivelinessLostStatus&)
+{
+    m_mainWindow->addMessageToOutput(QString("Lost liveliness"));
```

Comment:
```suggestion
    m_mainWindow->addMessageToOutput(QString("Writer lost liveliness"));
```
so there is no misunderstanding that it was the writer the entity that lost liveliness